pkgs/development/tools/ytt/default.nix

```diff
@@ -0,0 +1,32 @@
+{ lib, buildGoModule, fetchFromGitHub }:
+buildGoModule rec {
+  pname = "ytt";
+  version = "0.30.0";
+
+  src = fetchFromGitHub {
+    owner = "vmware-tanzu";
+    repo = "carvel-ytt";
+    rev = "v${version}";
+    sha256 = "0v9wp15aj4r7wif8i897zwj3c6bg41b95kk7vi3a3bzin814qn6l";
+  };
+
+  goPackagePath = "github.com/vmware-tanzu/carvel-ytt";
+
+  vendorSha256 = null;
+
+  # extracted from /hack/test-unit.sh
+  checkPhase = ''
+    go test "$(go list ./... | grep -v yaml.v2)"
+  '';
+
+  preFixup = ''
+    rm $out/bin/ytt-lambda-website
```

Comment:
Can we set ``subPackages = [ "cmd/something" ];`` to only build the required binary?

Reply:
Even without this change I get:

```
running tests
cannot find package "." in:
        /private/tmp/nix-build-ytt-0.30.0.drv-0/source/cmd/ytt
github.com/k14s/ytt/cmd/ytt-lambda-website
github.com/k14s/ytt/pkg/cmd
github.com/k14s/ytt/pkg/cmd/core
github.com/k14s/ytt/pkg/cmd/template
github.com/k14s/ytt/pkg/filepos
github.com/k14s/ytt/pkg/files
github.com/k14s/ytt/pkg/orderedmap
github.com/k14s/ytt/pkg/structmeta
github.com/k14s/ytt/pkg/template
github.com/k14s/ytt/pkg/template/core
github.com/k14s/ytt/pkg/texttemplate
github.com/k14s/ytt/pkg/version
github.com/k14s/ytt/pkg/website
github.com/k14s/ytt/pkg/workspace
github.com/k14s/ytt/pkg/yamlfmt
github.com/k14s/ytt/pkg/yamlmeta
github.com/k14s/ytt/pkg/yamltemplate
github.com/k14s/ytt/pkg/yttlibrary
github.com/k14s/ytt/pkg/yttlibrary/overlay
```

Reply:
Setting subPackages to `cmd/ytt` has the effect of only building the desired command. We can't run any tests this way, but overall should be faster.

I'm fine with doing it this way and only tried to include tests since they were erroring.